makes maxpool2d parameterizable

tsmcalister opened the pull request:
My try at making the maxpool2d layer parametrizable.

It is implemented by adding all **kwargs that are passed as properties to the ctx object.

geohot commented:
The kwargs thing is nice! Made a bunch of changes, want to rebase and I'll merge?

tsmcalister commented:
Ah, totally forgot about the default params!
I've implemented it with inspect.signature and rebased.
Also added the default param to the avg_pool2d layer.

geohot commented:
Sweet!
